### ISSUE TITLE
Secret Scanning not working on Linux agents

### ISSUE THREAD

**Issue by Argandov** [deepfence/ThreatMapper#2269]
# Technical Context
- **Components/Services affected**: Deepfence agent (Linux)
- **Affected Module:** Secret Scanning
- **OS version:** Debian GNU/Linux 12 (bookworm), AMD64
- **Deepfence agent version:** I have no idea; I am not familiar with the agents' architecture but the installer was `deepfence-agent-amd64-2.2.2.tar.gz`

## **Describe the bug**
1. The Secret Scanner is not working in Linux Debian agents. I get an "Error" without any context in Deepfence Web UI as shown below:

<img width="1197" alt="image" src="https://github.com/user-attachments/assets/797866ca-8a91-49af-8832-8cb0bceb69c2">

## What I've done to try to understand the error:

I am mostly grepping through the logs at `/opt/deepfence/df-agents/copilot/var/log/supervisor/deepfenced.log` 

<img width="1440" alt="image" src="https://github.com/user-attachments/assets/f16a54dd-4fb9-4d71-a317-86ec1a78c25c">

I am seeing WARN errors `skip <Deepfence_module> rules update already new` in the logs for deepfence agents in Linux. I don't know what that means, and I don't know either if there's something wrong with my installation (I don't know either if the WARN messages are related, or if this WARN errors may indicate some other issue I'm not even aware of yet, not related to secret scanning): 

<img width="1440" alt="image" src="https://github.com/user-attachments/assets/c62b3b44-1662-46c0-9a87-13eeb9b783ea">

# How I installed the agent:
1. `sudo bash`
2. `export MGMT_CONSOLE_URL="MY DF SERVER IP"`
     `export DEEPFENCE_KEY="MY KEY"` 
     (As per the instructions in the Web UI)
3. ` bash install_deepfence_agent.sh`

## **Expected behavior**
Secret Scanning on Linux hosts working 

Thank you, I don't know what to do here, if the error lies on my side or there's something wrong with DF-agents

**Comment by gnmahanth** [comment]
can you please show us the `deepfence-worker` logs and also is it possible to restart the management console and try again

**Comment by Argandov** [comment]
Hi, where are the deepfence-worker logs?

TL;DR: I restarted the management console. secret scanner logs on DF agent are empty. Same error again

<img width="1197" alt="image" src="https://github.com/user-attachments/assets/1bbcb2d0-3731-4a3d-addb-02f4768c4d27">


# Deepfence agent on Linux

I'm on the deepfence Linux agent and I don't know where are the `deepfence-worker` logs:

<img width="887" alt="image" src="https://github.com/user-attachments/assets/8b5806d0-f7e2-492d-b2c8-e44155681777">

---
the Secret_scanner.log file is empty (Even after restarting the management console and starting a secret scan again):

<img width="919" alt="image" src="https://github.com/user-attachments/assets/b6f077a9-0eb6-4f27-be39-5d11a4617f42">

# FTR: Restarting management console

I restarted the management console by:

`docker-compose -f docker-compose.yml down`

then up again:

`docker-compose -f docker-compose.yml up --detach`

**Comment by Argandov** [comment]
UPDATE:

I just realized you meant deepfence_worker Docker container. Here they are but I'm not sure what exactly would you like to see:

<img width="1419" alt="image" src="https://github.com/user-attachments/assets/5aa77ddc-06df-45ae-b795-72e0cffd7245">

<img width="1419" alt="image" src="https://github.com/user-attachments/assets/9dc2517f-6dc2-4a13-869c-6afa12da6399">

If you want me to do something else with the logs please tell me so I know what exactly to show you

**Comment by gnmahanth** [comment]
@Argandov 

I tried to run agent on debina 12 and I was able to run all the scans successfully
 
can you please provide the full log file `/opt/deepfence/df-agents/copilot/var/log/supervisor/deepfenced.log` we should be able to see why secret scanner is not starting

meanwhile can you try to uninstall and install the agent again 

if secret scanner start without any issues the secret scanner log file will have below content 
```
root@mahanth-debian-12:/opt/deepfence/df-agents/mahanth-debian-12/var/log/deepfenced# cat secret_scanner.log 
time="2024-07-26 04:16:05" level=error msg="open /opt/deepfence/df-agents/mahanth-debian-12/home/deepfence/bin/secret-scanner/config/config.yaml: no such file or directory"
time="2024-07-26 04:16:10" level=error msg="open /opt/deepfence/df-agents/mahanth-debian-12/home/deepfence/bin/secret-scanner/config/config.yaml: no such file or directory"
time="2024-07-26 04:16:15" level=error msg="open /opt/deepfence/df-agents/mahanth-debian-12/home/deepfence/bin/secret-scanner/config/config.yaml: no such file or directory"
INFO[2024-07-26T04:16:21Z] grpc.go:98 main: server listening at /opt/deepfence/df-agents/mahanth-debian-12/tmp/secret-scanner.sock
```
